monitoring/timedrift_test.go

```diff
@@ -0,0 +1,189 @@
+				node3.Name: s.clock.Now().Add(200 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, 200*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is acceptable, one node is lagging behind",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(-100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, -100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-3 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(500 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.failureProbe(node3, 500*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-2 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.successProbe(node3, 100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(time.Second),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.failureProbe(node3, time.Second),
+			},
+		},
+	}
+	for _, test := range tests {
+		clients, err := ttlmap.New(10)
+		c.Assert(err, check.IsNil)
+		for nodeName, nodeTime := range test.Times {
+			clients.Set(
+				nodes[nodeName].Addr.String(),
+				newMockedTimeAgentClient(nodeTime),
+				time.Hour)
+		}
+		checker := &timeDriftChecker{
+			TimeDriftCheckerConfig: TimeDriftCheckerConfig{
+				Clock: s.clock,
+			},
+			FieldLogger: logrus.WithField(trace.Component, "test"),
+			clients:     clients,
+			serfClient:  agent.NewMockSerfClient(test.Nodes, nil),
+			name:        node1.Name,
+			self:        node1,
```

Comment:
That's what I'm talking about. `node1` is anchor anyways, so it does not belong in any of the test cases.